src/main/java/kitchenpos/menu/tobe/domain/menu/validate/ProductValidator.java

```diff
@@ -0,0 +1,29 @@
+package kitchenpos.menu.tobe.domain.menu.validate;
+
+import kitchenpos.menu.tobe.domain.menu.MenuProduct;
+import kitchenpos.menu.tobe.domain.menu.ProductApiClient;
+import org.springframework.stereotype.Component;
+
+import java.util.List;
+
+@Component
+public class ProductValidator {
+    private final ProductApiClient productApiClient;
+
+    public ProductValidator(ProductApiClient productApiClient) {
+        this.productApiClient = productApiClient;
+    }
+
+    public void validateProductExistence(List<MenuProduct> menuProducts) {
+        long actualProductCount = productApiClient.countProductsByIds(
+                menuProducts.stream()
+                        .map(MenuProduct::getProductId)
+                        .toList()
+        );
```

Comment:
적절히 변수로 추출해주면 조금 더 가독성이 좋아질 것 같네요. 😃
```suggestion
        var productIds = menuProducts.stream()
            .map(MenuProduct::getProductId)
            .toList();

        long actualProductCount = productApiClient.countProductsByIds(productIds);
```